lib/irb/ruby-lex.rb

```diff
@@ -329,10 +326,16 @@ def check_code_block(code, tokens)
   def calc_nesting_depth(opens)
```

Comment:
Not directly related to this PR, but it seems `indent_level` and `nesting_level` are always used separately. Given that `nesting_level`'s condition should be much simpler and is only used in `#prompt`, should we split their implementation?

Also, it seems weird that `RubyLex#prompt` passes `nesting_level` to `@prompt`, but in `set_prompt` it names that argument as `indent`.

Reply:
I thought the implementation is similar so I combined it together.
But yes, splitting seems better now. I'll do it :+1: